sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -688,6 +691,15 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
+    private final Distribution backlogBytes = Metrics.distribution("io", "backlog.bytes");
+    private final Distribution backlogMessages = Metrics.distribution("io", "backlog.messages");
```

Comment:
Joining in on the `Gauge` suggestion. 
Sounds like something I could use in #2073 which is broken at the moment, and can be fixed either by resorting to the good old aggregators, or introducing a `Gauge`. 

I would be reluctant to introduce new code using the deprecated aggregators mechanism, so definitely 👍  for the `Gauge` metric for (possibly) saving the day.